libsolidity/interface/CompilerStack.cpp

```diff
@@ -464,9 +469,13 @@ bool CompilerStack::analyze()
 	return !m_hasError;
 }
 
-bool CompilerStack::parseAndAnalyze()
+bool CompilerStack::parseAndAnalyze(State _stopAfter)
```

Comment:
This function is really simple. But I'm having more a problem with the naming `parseAndAnalyse` and the `stopAfter` to break out early (skip analyse and only parse).

In general I think having a `stopAfter` is good, that could be applied at `compile(...)` level instead,
where the `compile()` function makes sure it stops after the right stage. If some other code path needs to explicitly parse but not analyze, I think it shouldn't call `parseAndAnalyze(CompilerStack::Parsed)` but rather call just `parse()` instead?

Reply:
I see your point, but `stopAfter` is more generic than a simple "stop after parsing" as it could refer to any stage. Considering this, `parseAndAnalyze()` can be seen as a certain subset of actions to be done that can be further limited using the parameter (which could also have a stage in the middle of analyse for example).

Shifting that logic to outside the call also adds more redundancy as the same code has to be moved to various places.

Reply:
We can have another PR that removes this function (from the public interface) such that we only have `compile()` and  current calls to `parseAndAnalize` are replaced by `compile(StopAfter::Analysis)`